src/test-run/index.js

```diff
@@ -155,10 +158,44 @@ export default class TestRun extends Session {
     }
 
     _resolvePendingRequest (command) {
+        this.lastDriverStatusResponse = command;
         this.pendingRequest.resolve(command);
         this.pendingRequest = null;
     }
 
+    _handleDriverRequest (driverStatus) {
+        if (!this.running)
+            this._start();
+
+        var shouldRejectPendingDriverTask = driverStatus.pageError &&
+                                            this.pendingDriverTask &&
+                                            isCommandRejectableByPageError(this.pendingDriverTask.command);
+
+        if (shouldRejectPendingDriverTask)
+            this._rejectPendingDriverTask(driverStatus.pageError);
+        else {
+            this.pendingJsError = this.pendingJsError || driverStatus.pageError;
+
+            if (this.pendingDriverTask) {
+                if (!driverStatus.isCommandResult)
+                    return this.pendingDriverTask.command;
+
+                if (isTestDoneCommand(this.pendingDriverTask.command)) {
+                    this.pendingDriverTask.resolve();
+
+                    return TEST_DONE_CONFIRMATION_COMMAND;
```

Comment:
`TEST_DONE_CONFIRMATION_RESPONSE`? Because, if it's a command we should use command ctor then. BTW, maybe we should add some data to this object for debugging purposes? Like returning string `test-done-confirmation` instead of object?